instance/management/commands/instance_statistics_csv.py

```diff
@@ -0,0 +1,294 @@
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_elasticsearch_data playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(
+                os.path.dirname(playbook_path),
+                'requirements.txt'
+            ),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'start_date: {start_date}\n'
+                'end_date: {end_date}'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix,
+                start_date=start_date,
+                end_date=end_date
+            ),
+            playbook_path=playbook_path,
+            username=settings.OPENSTACK_LOGS_SERVER_SSH_USERNAME,
+            logger_=log_line,
+        )
+
+    def get_appserver_statistics(self, playbook_output_dir, name_prefix, public_ip):
+        """ Execute the collect_appserver_data playbook to gather statistics """
+        inventory = '[apps]\n{server}'.format(server=public_ip)
+        playbook_path = os.path.join(
+            settings.SITE_ROOT,
+            'playbooks/collect_activity/collect_activity.yml'
+        )
+
+        def log_line(line):
+            """Helper to pass to capture_playbook_output()."""
+            self.stderr.write(self.style.SUCCESS(line))
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_appserver_data playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(os.path.dirname(playbook_path), 'requirements.txt'),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'local_output_filename: user_statistics\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'extra_script_arguments: --skip-hit-statistics'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix
+            ),
+            playbook_path=playbook_path,
+            username=settings.OPENSTACK_SANDBOX_SSH_USERNAME,
+            logger_=log_line,
+        )
+
+    def collect_instance_statistics(
+            self,
+            out,
+            domain_name,
+            start_date,
+            end_date
+    ):  # pylint: disable=too-many-locals
+        """Generate the instance statistics CSV."""
+        instance = self.get_instance_from_domain_name(domain_name)
+        appserver = instance.get_active_appservers().first()
```

Comment:
@pcockwell, note that for instances with more than 1 active appserver, this will make the returned value for unique hits incorrect. But that's okay because we get the more accurate data from elasticsearch anyway.

Reply:
This isn't relevant because the hits data isn't collected from the appserver for this command. Only the course and user counts are. See https://github.com/open-craft/opencraft/pull/597/files#diff-9a92f541e079630853c6563a8c3c48bfR224